File: leetcode/medium/2486. Append Characters to String to Make Subsequence/main.go
package main

func appendCharacters(s string, t string) int {
  i, j := 0, 0
  
  for i < len(s) && j < len(t) {
    if s[i] == t[j] {
      j++
    }

    i++
  }

  return len(t) - j
}

File: leetcode/medium/2486. Append Characters to String to Make Subsequence/main.rs
impl Solution {
    pub fn append_characters(s: String, t: String) -> i32 {
        let mut t = t.bytes().peekable();

        s.bytes().for_each(|c| {
            t.next_if_eq(&c);
        });

        t.len() as _
    }
}
